### DISCUSSION TITLE
VRP: add multiple time window

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#2717] [Routing (and legacy CP) questions] [answered]
I am wondering how to configure the model to add multiple time window for one pickup or delivery point. i.e. a customer can have three time windows (8:00-10:00, 14:00-16:00, and 19:00-20:00) available for receiving the goods. 

```python
	for location_idx, time_window in enumerate(data.time_windows):
		if location_idx == start_node or location_idx == end_node:
			continue
		index = manager.NodeToIndex(location_idx)
		time_dimension.CumulVar(index).SetRange(time_window[0], time_window[1])
		routing.AddToAssignment(time_dimension.SlackVar(index))
```

Do we simply add three more lines of above codes?

**Comment by Mizux** [comment, marked as answer]
setRange will "reset" the entire range so you can't use it three time, but you can use RemoveIntervals()...

Please take a look at https://github.com/Mizux/tsp_multiple_days/blob/7201cf2017bd89bd99becd83375bc40e177d2b72/40_Locations_3_days.py#L102-L118

**Reply by Westlife1002** [reply]
thank you very much. I will read through the suggested solution.
